confidential/confidential.go

```diff
@@ -91,3 +91,34 @@ func FinalValueBlindingFactor(input FinalValueBlindingFactorInput) ([32]byte, er
 
 	return secp256k1.BlindGeneratorBlindSum(ctx, values, generatorBlind, blindingFactor, len(input.InValues))
 }
+
+//AssetCommitment method generates asset commitment
+func AssetCommitment(asset []byte, factor []byte) ([33]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	generator, err := secp256k1.GeneratorGenerateBlinded(ctx, asset, factor)
+	if err != nil {
+		return [33]byte{}, err
+	}
+
+	return secp256k1.GeneratorSerialize(ctx, generator), nil
+}
+
+//ValueCommitment method generates value commitment
+func ValueCommitment(value uint64, generator []byte, factor []byte) ([33]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	gen, err := secp256k1.GeneratorParse(ctx, generator)
+	if err != nil {
+		return [33]byte{}, err
+	}
+
+	commit, err := secp256k1.Commit(ctx, factor, value, gen)
+	if err != nil {
+		return [33]byte{}, err
+	}
+
+	return secp256k1.CommitmentSerialize(ctx, commit)
```

Comment:
same here, also the `*Commitment` type of the *go-secp256k1-zkp* lib has an available  `Bytes()` method:
```suggestion
	return commit.Bytes(), nil
```

Reply:
Done